com.avaloq.tools.ddk.xtext.builder/src/com/avaloq/tools/ddk/xtext/builder/FingerprintResourceDescription.java

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Avaloq Evolution AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Avaloq Evolution AG - initial API and implementation
+ *******************************************************************************/
+
+package com.avaloq.tools.ddk.xtext.builder;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.InternalEObject;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.xtext.naming.QualifiedName;
+import org.eclipse.xtext.resource.EObjectDescription;
+import org.eclipse.xtext.resource.IEObjectDescription;
+import org.eclipse.xtext.resource.IReferenceDescription;
+import org.eclipse.xtext.resource.IResourceDescription;
+import org.eclipse.xtext.resource.impl.AbstractResourceDescription;
+import org.eclipse.xtext.resource.impl.EObjectDescriptionLookUp;
+
+import com.avaloq.tools.ddk.xtext.resource.IFingerprintComputer;
+import com.avaloq.tools.ddk.xtext.resource.PatternAwareEObjectDescriptionLookUp;
+import com.avaloq.tools.ddk.xtext.resource.extensions.IResourceDescription2;
+import com.google.common.collect.ImmutableList;
+
+
+/**
+ * Lightweight resource description that contains only the data necessary for the invalidations.
+ * That is, the resource URI and a list of exported objects' descriptions with their fingerprints.
+ */
+public class FingerprintResourceDescription extends AbstractResourceDescription implements IResourceDescription2 {

```

Comment:
I think this needs more documentation about the assumptions. I.e. you assume EClass etc. to be part of the fingerprint. You also actually assume fingerprints to be used.